test/validation.html

```diff
@@ -0,0 +1,155 @@
+      </vaadin-date-time-picker>
+    </template>
+  </test-fixture>
+
+  <script>
+    customElements.whenDefined('vaadin-date-time-picker').then(() => {
+      class DateTimePicker2020Element extends customElements.get('vaadin-date-time-picker') {
+        checkValidity() {
+          return this.value === '2020-02-02T20:20';
+        }
+      }
+      customElements.define('vaadin-date-time-picker-2020', DateTimePicker2020Element);
+    });
+  </script>
+
+  <test-fixture id="custom-validation">
+    <template>
+      <vaadin-date-time-picker-2020></vaadin-date-time-picker-2020>
+    </template>
+  </test-fixture>
+
+  <script>
+    describe(`Validation`, () => {
+      let dateTimePicker;
+      let customField;
+
+      ['default', 'slotted'].forEach(set => {
+
+        describe(`Default (${set})`, () => {
+
+          beforeEach(() => {
+            if (set === 'default') {
+              dateTimePicker = fixture('default');
+              customField = dateTimePicker.$.customField;
+            } else if (set === 'slotted') {
+              dateTimePicker = fixture('slotted-inputs');
+              customField = dateTimePicker.$.customField;
+            }
+          });
+
+          afterEach(() => {
+            dateTimePicker.invalid = false;
+            dateTimePicker.errorMessage = '';
+          });
+
+          it('should be required', () => {
+            dateTimePicker.required = true;
+            expect(customField.required).to.be.true;
+          });
+
+          it('should not be required', () => {
+            expect(customField.required).to.be.false;
+          });
+
+          it('should validate correctly with required flag', done => {
+            dateTimePicker.name = 'foo';
+            dateTimePicker.required = true;
+
+            setTimeout(() => {
+              expect(dateTimePicker.validate()).to.equal(false);
+              expect(dateTimePicker.invalid).to.be.equal(true);
+
+              dateTimePicker.value = '2020-02-02T02:02:00';
+              expect(dateTimePicker.validate()).to.equal(true);
+              expect(dateTimePicker.invalid).to.be.equal(false);
+
+              done();
+            });
+          });
+
+          it('should validate min/max dates', () => {
+            dateTimePicker.min = '2020-02-02T02:00';
+            dateTimePicker.max = '2020-02-02T04:00';
+
+            // Set invalid value.
+            dateTimePicker.value = '2020-02-02T01:00';
+            expect(dateTimePicker.validate()).to.equal(false);
+            expect(dateTimePicker.invalid).to.be.equal(true);
+
+            dateTimePicker.value = '2020-02-02T03:00';
+            expect(dateTimePicker.validate()).to.equal(true);
+            expect(dateTimePicker.invalid).to.be.equal(false);
+          });
+
+          it('should be possible to force invalid status', () => {
+            dateTimePicker.invalid = true;
+            expect(customField.invalid).to.be.true;
+          });
```

Comment:
`vaadin-text-field` has a different logic for validation and I would say it's unique for our components. It will behave differently if you follow:
```
textField.maxlength = 10;
textField.invalid = true;
textField.validate() --> makes the field valid
```
`text-field` is checking the existence of specific validation constraints or returning `!this.invalid` on `checkValidity`. Try also taking a look onto `vaadin-time-picker`.

I guess we shouldn't align `vaadin-date-time-picker` and `vaadin-text-field` as `date-picker` and `time-picker` don't have logic for verification of validation constraints, so the containing component shouldn't have either.

However the `date-picker` indeed has some probs with validation and I've already created a ticket some time ago: https://github.com/vaadin/vaadin-date-picker/issues/696

Reply:
I created another issue https://github.com/vaadin/vaadin-date-picker/issues/697

I guess `validate()` is fine as it is then, but unfocus/blur should not cause the explicitly set invalidity to be removed. This is currently consistent with both vaadin-text-field and vaadin-time-picker, but not in vaadin-date-picker (above issue).

Reply:
nvm I think I was confusing DTP and DP for a while. DTP doesn't remove invalid now on blur (good) if it has been set explicitly so it's only a bug in DP.